File: cpp/p22.cpp
#include "euler.h"

using namespace std;

int main() {
  ifstream file( "../data/names.txt" );
  string line;
  vector<string> names;
  int total = 0;
	
  while( getline( file, line ) ) {
    vector<string> new_names = tokenize( line, ", \"" );
    names.insert( names.end(), new_names.begin(), new_names.end() );
  }

  sort( names.begin(), names.end() );
	
  for( int i = 0; i < names.size(); i++ ) {
    for( int j = 0; j < names[ i ].size(); j++ ) 
      total += ( names[ i ][ j ] - 64 ) * ( i + 1 );
  }
	
  cout << total << endl;
}

File: clojure/p22.clj
(ns p22
  (:use [euler :only (word-value read-quoted-csv sum timed-test)])
  (:require [clojure.string :as str]))

(defn name-score [idx name]
  (* (inc idx) (word-value name)))

(timed-test
 871198282
 (let [names (read-quoted-csv "../data/names.txt")]
   (sum (map-indexed name-score (sort names)))))
